azure-pipelines.yml

```diff
@@ -242,53 +242,48 @@ jobs:
       versionSpec: '$(python.version)'
 
   - bash: |
+      set -e
       sed -i "s/'giotto-tda'/'giotto-tda-nightly'/1" setup.py
       sed -i "s/__version__.*/__version__ = '$(Build.BuildNumber)'/1" gtda/_version.py
-    failOnStderr: true
     condition: eq(variables['nightly_check'], 'true')
     displayName: 'Change name to giotto-tda-nightly'
 
   - script: |
       python -m pip install --upgrade pip setuptools
       pip install -e ".[tests, doc]"
-    failOnStderr: true
     displayName: 'Install dev environment'
 
   - script: |
-      pytest gtda --cov --cov-report xml
+      pytest gtda --cov --cov-report xml || exit /b
       flake8
```

Comment:
Here are we combining the explicit erroring in the line above with the fact that we expect Azure to fail on standard error for the `flake8` line, despite the removal of `failOnStderr: true`?

Reply:
Indeed. I would move flake8 into a separate job as well, as there is no need to check it for all OSs.

Reply:
Great, will do this in another PR.